index.js

```diff
@@ -366,9 +368,23 @@ Archive.prototype.appendFile = function (filename, name, cb) {
       mode: st.mode,
       size: 0,
       link: null
-    }, {filename: filename}, cb)
+    }, {filename: filename}, function (err, entry) {
+      if (err) return cb(err)
+      if (!entry) return cb()
+      if (entry.type === 'file') self.readStats.files++
+      cb(err, entry)
+    })
 
-    if (ws) pump(fs.createReadStream(path.resolve(self.directory, filename)), ws)
+    var statsCounter = through(function (data, encoding, next) {
+      self.readStats.totalRead += data.length
+      this.push(data)
+      next()
+    })
+
+    if (ws) {
+      var readStream = fs.createReadStream(path.resolve(self.directory, filename))
+      pump(readStream, statsCounter, ws)
+    }
```

Comment:
@joehand I think you'll need to return something immediately from `appendFile` that has the stats object attached to it. @mafintosh what do you think the API should be for that?

Also I think its OK hyperdrive has the stats tracking code in it, since other hyperdrive users beyond just dat would probably use it. I think dat should handle formatting the stats the printing them out, hyperdrive should just collect the stats as simply as possible and make them available. For things like averaging etc or doing the total sum of all files its prob easier to do it in dat right now.